### PR TITLE
Un-namespace Python packages

### PR DESCRIPTION
**Pull request by soltanmm-google**
Part 1 of a fix to being horribly and utterly broken w/ the setuptools update that we'll eventually want to duplicate on `v1.0.x`.

This breaks everything but `grpcio`.

cc @ctiller

---

Failures: #8989, #9082

### CODE REVIEW

**Comment by soltanmm-google** [comment]
(clarification: Parts 1+ℕ will be to reduce the impact on users of `grpcio-*` distributions)

**Comment by soltanmm-google** [comment]
This shouldn't include those, I don't think. This should just green the build, and whatever parts we decide to paper-over the user-experience bumps can come after.

**Comment by soltanmm-google** [comment]
And thus I eat my words; updated to paper over loss of namespaced packages.

File: src/python/grpcio/grpc/__init__.py [hunk 1]
```diff
@@ -31,12 +31,33 @@
...
 import abc
 import enum
+import sys
 
 import six
 
 from grpc._cython import cygrpc as _cygrpc
 
 
+############################### Extension Shims ################################
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/__init__.py]: Drop this; we don't really want to help readers navigate to this part of the code.

File: src/python/grpcio/grpc/__init__.py [hunk 2]
```diff
@@ -31,12 +31,33 @@
...
+
+try:
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/__init__.py]: Add a comment to the effect of "these are here to maintain compatibility; do not use!"?

File: src/python/grpcio/grpc/__init__.py [hunk 3]
```diff
@@ -31,12 +31,33 @@
...
+  import grpc_tools
+  sys.modules.update({'grpc.tools': grpc_tools})
+except ImportError:
+  pass
+
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/__init__.py]: Drop these blank lines; this is one code block.

File: src/python/grpcio/grpc/__init__.py (hunk 2 above)
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/__init__.py]: ... and yeah, let's move these to the very end of the file.

**Comment by soltanmm-google** [comment]
addressed sans stuff discussed offline

**Pull request merged**